### ISSUE TITLE
Self-Follow / Self-Reference in Repo is possible

### ISSUE THREAD

**Issue by proggeramlug** [bluesky-social/atproto#1849]
**Describe the bug**

It is not noticeable when using most clients. But it is actually possible to follow oneself or to put it in other terms: To have a self-reference in a repository. There might be cases where this is actually desired but in terms of followers, it most likely is not.

**To Reproduce**

1. Send a request to the "/xrpc/com.atproto.repo.createRecord" endpoint with the same `record.subject` as the `repo` key. It will succeed.
2. Check the followers then. Either via API or via client (same anyways). It did notice that not all followers are immediately visible - a new account of mine was showing "2 followers" but only one came back via API. Probably unrelated to this issue.
3. Example: https://bsky.app/profile/amlug.net/followers This, my main account, follows itself.

**Expected behavior**

Reject self-reference for most repo's, at the very least for followers.

**Details**

- Operating system: Any
- Node version: Any

**Comment by snarfed** [comment]
Out of curiosity, does a self-follow cause any actual problems anywhere?

**Comment by proggeramlug** [comment]
Not to my knowledge, it just inflates the follower number by 1.

**Comment by bnewbold** [comment]
Yup! You can also "like" your own content.

We'll eventually handle in the indexing code.

**Comment by proggeramlug** [comment]
Agree, but is liking your own stuff a self-reference the same way a follow is?
